Dashboard Fixes

mayadomi opened the pull request:
The dashboards are as functional as I can get them :D 

All that's missing that probs ought to be added in is the auth check for is_staff to handle what shows more gracefully than blank boxes.

Go nuts with CSS if you have time - it's verrrrry basic atm.

Hopefully it all works correctly with the backend!

mayadomi committed:
Bones of the update/solution for single-button assigning of mentors

Tidy up/extend to remove mentors next.

mayadomi committed:
Dashboards are go!

I think everything is working. Just need to add auth checks for staff.

mayadomi committed:
Forgot a data scenario (ooooops)

Forgot logic to show button to assign if event_mentors table is
empty/fresh. Also add some extra text to hint at why might not
see a mentor (ie they only show up if they're of status ready)

mayadomi committed:
Adding auth checks for admin dashboard
So that only staff can access and if not staff a custom view.
Also shifted the top dashboard components to pages, as these are kind of what they are.

mayadomi committed:
Tidying up CSS :D

rosiejeays committed:
Pulling skills from database

rosiejeays committed:
Adding unconfirmed/all mentor events

rosiejeays committed:
Pulling skills from database

rosiejeays commented on src/components/EditEventsForm/EditEventsForm.jsx:
```diff
@@ -25,143 +24,133 @@ function EditEventForm({editEventOpen, onEditEventClick, activeEvent, onChangeAc
-    // }   else {
-    //     setFormInvalid("Must be staff to create an event")
-    // }
+        // }   else {
```
To be removed?

rosiejeays commented on src/components/EditEventsForm/EditEventsForm.jsx:
```diff
@@ -25,143 +24,133 @@ function EditEventForm({editEventOpen, onEditEventClick, activeEvent, onChangeAc
+    }, [events, activeEvent])
+
     const handleChange = (event) => {
         // if (auth.token){
```
To be removed?

rosiejeays commented on src/components/EditEventsForm/EditEventsForm.jsx:
```diff
@@ -25,143 +24,133 @@ function EditEventForm({editEventOpen, onEditEventClick, activeEvent, onChangeAc
-        
         event.preventDefault()
-
         // if (auth.token){
```
to be removed?

rosiejeays commented on src/components/EditEventsForm/EditEventsForm.jsx:
```diff
@@ -25,143 +24,133 @@ function EditEventForm({editEventOpen, onEditEventClick, activeEvent, onChangeAc
+        } else {
+            setFormInvalid("Please complete the form")
+        }
         // } else {
```
to be removed?

rosiejeays commented on src/components/EventCrud/EventCrud.jsx:
```diff
@@ -1,30 +1,33 @@
-    const onCurrentMentorsChange = (currentMentorsList) => {
-        setCurrentMentors(currentMentorsList)
+    
+    //TODO should be utils
```
move this to utilities - there's a lot going on in these files and if it's being reused that's a win.

mayadomi replied:
Agree! Will migrate to your utils one at some stage. The only bit that will need the current way is in Edit Event Form, where the date-picker wants dash-delimited date (in order to display previous date that user is editing)

rosiejeays commented on src/components/MentorList/MentorList.jsx:
```diff
@@ -1,98 +1,85 @@
-    // Some of these should be grabbed from API (skills)
-    const skills = ["Python", "Django", "DRF", "React", "Javascript", "Front-end", "Back-end", "HTML-CSS"]
+    
+    // TODO Some of these should be grabbed from API (skills)
```
we can probably remove this comment now that we're grabbing skills from the database

mayadomi replied:
Thanks for switching these over! :D

rosiejeays commented on src/components/MentorListDetails/MentorListDetails.jsx:
```diff
@@ -1,68 +1,139 @@
-    // const [searchTermRank, setSearchTermRank] = useState("")
+    const { skills, skillsLoading, skillsError } = useSkills([]);
+
+    // TODO Some of these should be grabbed from API (skills)
```
This can be removed

rosiejeays commented on src/components/MentorListDetails/MentorListDetails.jsx:
```diff
@@ -1,68 +1,139 @@
+
+    useEffect(() => {
+
+        // Filtering the mentor list based on chosen selection in any of three filters - location/rank/skill
```
Are we doing this in filter in multiple places? Should it be pulled out as a utility?

mayadomi replied:
Eventually I reckon yes, as it's used (slightly differently) in two places - but I've put that in the post-MVP bucket of things to do 😆

rosiejeays commented on src/components/NavBar/NavBar.jsx:
```diff
@@ -103,6 +105,36 @@ function NavBar() {
                             <Link to="/" onClick={handleLogout}>
                                 Log Out
                             </Link>
+                            
```
This doesn't immediately update - if I log in as a non staff, then log in as staff I don't see these in the nav bar until I refresh. Same as if I've been logged in as staff then log in as normal user, I see these in the nav bar until I refresh

mayadomi replied:
Yeah I had the same - I suspect there's another bit of logic I need to add somewhere.

mayadomi replied:
Fixed by:
Set redirect in post-login to profile page
Use the profile page 'get self' to check if is_staff=true
If so, update auth (setAuth)
Point the nav bar check at this state's property

rosiejeays commented on src/components/CreateEventForm/CreateEventForm.jsx:
```diff
@@ -24,137 +24,122 @@ function CreateEventForm({createEventOpen, onCreateEventClick, activeEvent, onCh
 
     const handleChange = (event) => {
         // if (auth.token){
```
maybe remove the unused commented out code

mayadomi replied:
Done :D

rosiejeays commented on src/components/MentorListDetails/MentorListDetails.jsx:
```diff
@@ -1,68 +1,139 @@
+                            return a.first_name - b.first_name
+                        }).map((mentorDataDetails, key) => {
+
+                            // TODO These should be a card component
```
Agreed :)

mayadomi replied:
Might get to this, otherwise it's a post-preso task!

mayadomi replied:
Ended up doing it 😆

rosiejeays commented on src/components/MentorToEventsList/MentorToEventsList.jsx:
```diff
@@ -1,178 +1,155 @@
+            } 
+        }
+
         //mentor previously assigned, update confirmed to true
```
Loving these comments explaining what the code is doing

rosiejeays commented on src/pages/EventDashboard/EventDashboard.jsx:
```diff
@@ -0,0 +1,74 @@
+                    }
+                </div>
+                :
+                <div className='forbidden'>
```
This looks ridiculously amazing, however, we do already have a 404 page.
I recommend either:
a) using the existing 404 page or
b) updating the existing 404 page to have this content instead, then call on this 404 page component here

mayadomi replied:
Will swap to call the existing 404 - and we can check in with team on how we want it to look :D

rosiejeays commented on src/pages/EventDashboard/EventDashboard.jsx:
```diff
@@ -0,0 +1,74 @@
+                    <h3>Sorry fair user, ye must be of staff origins to access this page</h3>
+                    <div className='forbidden-lands'></div>
+                    <div className='compass'><img src='/compass.png'></img></div>
+                    {/* <a target="_blank" href="https://icons8.com/icon/kJqq7olgcblR/compass">Compass</a> icon by <a target="_blank" href="https://icons8.com">Icons8</a> */}
```
Is this getting used?

mayadomi replied:
Have reverted to 404 page :)

rosiejeays commented on src/pages/MentorDashboard/MentorDashboard.jsx:
```diff
@@ -0,0 +1,56 @@
+                <img  className='compass'src='/compass.png'></img>
+                <div className='forbidden-lands'></div>
+                
+                {/* <a target="_blank" href="https://icons8.com/icon/kJqq7olgcblR/compass">Compass</a> icon by <a target="_blank" href="https://icons8.com">Icons8</a> */}
```
Is this getting used?

rosiejeays commented on src/components/MentorList/MentorList.jsx:
I'm not exactly sure where or why, but one of my users comes up with multiple Assign buttons.
As chatted about in slack - this user is only registered to one event and there are no duplicates in the eventmentors table on my local db
![image](https://github.com/SheCodesAus/layer_8_victory_frontend_bris_2023/assets/116546864/1ebf4611-c1c0-41cd-b41b-c779937299a6)

mayadomi replied:
When you have a mo, send through your SQLite db; can't seem to replicate this. I'm sure it's some data nuance haven't accounted for :D

rosiejeays replied:
It was user 1, event 15 I believe
<img width="1385" alt="Screenshot 2023-10-27 at 8 04 28 am" src="https://github.com/SheCodesAus/layer_8_victory_frontend_bris_2023/assets/116546864/4359393d-fc71-4de2-a362-f54bdf920b1c">
<img width="1394" alt="Screenshot 2023-10-27 at 8 06 19 am" src="https://github.com/SheCodesAus/layer_8_victory_frontend_bris_2023/assets/116546864/1b09f09a-78bb-4443-a31e-5ddecaf7b70a">
<img width="1118" alt="Screenshot 2023-10-27 at 8 03 38 am" src="https://github.com/SheCodesAus/layer_8_victory_frontend_bris_2023/assets/116546864/2eaa62f1-d0c1-46ba-b979-ec4e178ce6a0">

mayadomi replied:
Fixed by removing the sort in MentorList.jsx that was calling the MentorCards - if first name/last name was same or missing we got funky behaviours 💃

rosiejeays reviewed (changes requested):
This is phenomenal effort.
A couple of things to fix up - let me know how I can help :D

mayadomi committed:
PR related changes
Mostly removing commmented out code and swapping non-staff users
to 404 page if they try to access dashboards.

mayadomi committed:
Updated CSS for Events Dashboard
Also tweaked button logic in the MentorCard to use 'find' instead
of 'filter'.

Andrea-Rivera reviewed (approved):
You did an amazing job Maya. 
From what I have tested, everything is working.
I think that once approved we can still work on the development branch.

mayadomi committed:
MentorDetailCard added, bulk CSS, NavBar update

mayadomi committed:
Merge branch 'development' into dashboard_tweaks

mayadomi committed:
Forgot CSS for create/edit event; fixed last issue mentor MentorCard

Date function swap & other merge changes :)

rosiejeays commented on src/components/AuthProvider.jsx:
```diff
@@ -10,6 +10,7 @@ export const AuthProvider = (props) => {
     const [auth, setAuth] = useState({
 //      // Here we initialize the context with the token from local storage, this way if the user refreshes the page we can still have the token in memory.
         token: window.localStorage.getItem("token"),
+        is_staff: "false"
```
Is this the fix for the navbar?

rosiejeays replied:
It appears to cause the navbar to reload without the admin navbar links after a short period of time :(
Perhaps we should store an is_staff variable in the local storage - like we do the token - and use that to ensure the navbar is updated correctly from there??

rosiejeays commented on src/components/CreateEventForm/CreateEventForm.css:
```diff
@@ -40,6 +44,11 @@
     justify-content: space-between;
     align-items: center;
     width: 100%;
+
```
maybe remove the extra line

rosiejeays commented on src/components/CreateEventForm/CreateEventForm.css:
```diff
@@ -62,24 +75,40 @@
+    background-color: var(--dark-blue);
+    color: var(--cream);
+    font-size: 15px;
+
```
and this one ;)

rosiejeays commented on src/components/EditEventsForm/EditEventsForm.jsx:
```diff
@@ -77,13 +75,16 @@ function EditEventForm({ editEventOpen, onEditEventClick, activeEvent, onChangeA
-        return formattedDate.toLocaleDateString("sv-SE")
-    }
+    // Swapping this to new utils and adding extra spice
+
```
maybe these extra spaces too

rosiejeays commented on src/components/MentorListDetails/MentorListDetails.jsx:
```diff
@@ -19,10 +20,10 @@ function MentorListDetails({ onEditMentorClick, editMentorOpen, onChangeActiveMe
-        onChangeActiveMentor(event.target.value)
-        onEditMentorClick("true")
-    }
+    // const handleOpenMentor = (event) => {
```
Maybe remove this if it's no longer being used

rosiejeays reviewed (approved):
This is unbelievably amazing work. Well done.
I've left some suggestions re extra enter lines or commented out code that could be removed.

The navbar is still buggy (see comment on that file), but I'm happy for this one to be merged into development as is. We can always review the navbar in another branch.

SHIP ITTTTTTT 🚢
